src/extensions/default/NavigationAndHistory/main.js

```diff
@@ -0,0 +1,620 @@
+        // to the persisted paneId when specified and we are in split view or unable to determine the paneid
+        var activePaneId = MainViewManager.getActivePaneId(),
+            targetPaneId = contextData.paneId; // Assume we are going to use the last associated paneID
+
+        // Detect if we are not in split mode
+        if (MainViewManager.getPaneCount() === 1) {
+            // Override the targetPaneId with activePaneId as we are not yet in split mode
+            targetPaneId = activePaneId;
+        }
+
+        // If hide of MROF list is a context parameter, hide the MROF list on successful file open
+        if (contextData.hideOnOpenFile) {
+            _hideMROFList();
+        }
+
+        return CommandManager
+            .execute(Commands.FILE_OPEN,
+                    {   fullPath: contextData.path,
+                        paneId: targetPaneId
+                    }
+                )
+            .done(function () {
+                if (contextData.cursor) {
+                    activeEditor = EditorManager.getActiveEditor();
+                    activeEditor.setCursorPos(contextData.cursor);
+                    activeEditor.centerOnCursor();
+                }
+            });
+    }
+    
+    /**
+     * Creates an entry for MROF list
+     * @private
+     * @param {String} path - full path of a doc
+     * @param {String} pane - the pane holding the editor for the doc
+     * @param {Object} cursorPos - current cursor position
+     * @return {Object} a frame containing file path, pane and last known cursor
+     */
+    function _makeMROFListEntry(path, pane, cursorPos) {
+        return {
+            file: path,
+            paneId: pane,
+            cursor: cursorPos
+        };
+    }
+    
+    /**
+     * Determines if a file is dirty
+     * @private
+     * @param {!File} file - file to test
+     * @return {boolean} true if the file is dirty, false otherwise
+     */
+    function _isOpenAndDirty(file) {
+        // working set item might never have been opened; if so, then it's definitely not dirty
+        var docIfOpen = DocumentManager.getOpenDocumentForPath(file.fullPath);
+        return (docIfOpen && docIfOpen.isDirty);
+    }
+    
+    /** 
+     * Returns a 'context' object for getting/setting project-specific preferences 
+     */
+    function _getPrefsContext() {
+        var projectRoot = ProjectManager.getProjectRoot();
+        return { location : { scope: "user", layer: "project", layerID: projectRoot && projectRoot.fullPath } };
+    }
+    
+    
+    function _checkExt(entry, index) {
+        var deferred = new $.Deferred(),
+            fileEntry = FileSystem.getFileForPath(entry.file);
+
+        fileEntry.exists(function (err, exists) {
+            if (!err && exists) {
+                deferred.resolve();
+            } else {
+                _mrofList[index] = null;
```

Comment:
I haven't tested this, and I guess it still behaves fine, but wouldn't it be more senseful to remove the entry from the array altogether?
`_mrofList.slice(index, 1)`